securedrop_client/gpg.py

```diff
@@ -0,0 +1,50 @@
+from pretty_bad_protocol import gnupg
+from uuid import UUID
+
+from securedrop_client.models import Source
+
+
+class GpgHelper:
+
+    def __init__(self, gpg_home: str, session) -> None:
+        self.gpg_home = gpg_home
+        self.gpg = gnupg.GPG(homedir=self.gpg_home)
+        self.session = session
+
+    def import_key(self, source_uuid: UUID, key_data: str) -> None:
+        local_source = self.session.query(Source).filter_by(uuid=source_uuid) \
+            .one_or_none()
+        if local_source is None:
+            raise RuntimeError('Local source not found: {}'
+                               .format(source_uuid))
+
+        res = self.gpg.import_keys(key_data)
+        if not res:
+            raise RuntimeError('Failed to import key.')
+
+        # using a set because importing a private key returns two identical
+        # fingerprints
+        fingerprints = set(res.fingerprints)
+        if len(fingerprints) != 1:
+            raise RuntimeError('Expected only one fingerprint.')
+
+        local_source.fingerprint = fingerprints.pop()
+        self.session.add(local_source)
+        self.session.commit()
+
+    def encrypt_to_source(self, source_uuid: UUID, message: str) -> str:
+        local_source = self.session.query(Source).filter_by(uuid=source_uuid) \
+            .one_or_none()
+        if local_source is None:
+            raise RuntimeError('Local source not found: {}'
+                               .format(source_uuid))
+
+        out = self.gpg.encrypt(message, local_source.fingerprint)
```

Comment:
There was nothing that seemed to reference this key anywhere, so I left that out for now.

Assuming we have a known path for the key, we can always reload it at boot time to store a reference to the fingerprint. Reloading a key will always provide a fingerprint.

```
>>> res = g.import_keys(k)
>>> res.fingerprints
['0CEC936888A60171461174C5C0A2586F09D77C82']
>>> res = g.import_keys(k)
>>> res.fingerprints
['0CEC936888A60171461174C5C0A2586F09D77C82']
```

I don't want to put it in the config because if a key gets rotated, we have to remember to rotate the config, and that sounds like a place where errors could happen.